src/wmtk/TriMeshOperationExecutor.cpp

```diff
@@ -367,6 +367,101 @@ void TriMesh::TriMeshOperationExecutor::replace_incident_face(
 }
 
 Tuple TriMesh::TriMeshOperationExecutor::split_edge()
+{
+    if (!m_mesh.multi_mesh_manager.is_parent_mesh()) {
+        return split_edge_single_mesh();
+    }
+    else
+    {
+        bool is_boundary_on_parent = m_mesh.is_boundary(m_operating_tuple);
+        
+        std::vector<Tuple> vec_t_child;
+        std::vector<Tuple> vec_t_opp_child;
+        vec_t_child = MultiMeshManager::map_edge_tuple_to_all_children(m_mesh, m_operating_tuple);
+        if (!is_boundary_on_parent)
+        {
+            Tuple t_opp = m_mesh.switch_edge(m_operating_tuple);
+            vec_t_opp_child = MultiMeshManager::map_edge_tuple_to_all_children(m_mesh, t_opp);
```

Comment:
there's no need to compute every version of this edge right now. just do it in the child_mesh_ptr loop and find a single one at a time